### DISCUSSION TITLE
Keycloack docs not fully complete?

### DISCUSSION THREAD

**Discussion by ralisebelcan** [Lissy93/dashy#409] [Help and Support] [answered]
Hello!
I've installed Dashy and was planning to to add it to keycloack, but get into the loop: I get redirected to SSO login page, pass my creds, but after that my request starts to travel between Keycloack and Dashy containers. Though I see that the session is established. Am I missing something or there should be some other Redirect URL passed to Keycloack? 
Haven't seen that info in documentation.

**Comment by lissy93** [comment, marked as answer]
Yeah, so when you set up the clients in the KC interface, there's a field called "Valid Redirect URIs" ([screenshot](https://user-images.githubusercontent.com/1862727/148599768-db4ee4f8-72c5-402d-8f00-051d999e6267.png)), put where you're hosting Dashy there. You may also need to set some headers, e.g. `Access-Control-Allow-Origin [URL-of Dashy]`
There's a guide on setting up Keycloak with Dashy in the [Authentication Docs](https://github.com/Lissy93/dashy/blob/master/docs/authentication.md#keycloak) which covers this in more detail.
Hope that helps :)
